pkg/controller/containersource/reconcile_test.go

```diff
@@ -284,71 +282,231 @@ var testCases = []controllertesting.TestCase{
 			}(),
 		},
 		IgnoreTimes: true,
-	}, /*
-		TODO(n3wscott): This does not work yet because we are only mocking the dynamic client
-		response and not the client list response. Fix this and the test will work.
-		{
-			Name:       "valid containersource, sink, and deployment",
-			Reconciles: &sourcesv1alpha1.ContainerSource{},
-			InitialState: []runtime.Object{
-				func() runtime.Object {
-					s := getContainerSource()
-					s.UID = containerSourceUID
-					return s
-				}(),
+	}, {
+		Name:       "valid containersource, sink, and deployment",
+		Reconciles: &sourcesv1alpha1.ContainerSource{},
+		InitialState: []runtime.Object{
+			func() runtime.Object {
+				s := getContainerSource()
+				s.UID = containerSourceUID
+				return s
+			}(),
+			func() runtime.Object {
+				// TODO(n3wscott): this is very strange, I was not able to get
+				// the fake client to return the resources.MakeDeployment version
+				// back in the list call. I might have missed setting some special
+				// metadata? Converting an unstructured and setting the fields
+				// I care about did work.
```

Comment:
I don't think mocking a response is necessary. After creating the deployment, the reconciler immediately returns, there is no situation when we need to reconcile a deployment that has just been created. From the perspective of the reconciler reacting to the Deployment, the Deployment already exists. It should be enough to add a Deployment object to the `InitialState` array, no mocks required.